### PR TITLE
add test files

### LINKED ISSUE #2939: installed MDAnalysisTests error/fail due to missing files
## Expected behavior ##

After a conda installation of 1.0.0 (with workaround, see #2938 ...), all tests should pass, as documented.

## Actual behavior ##

5 failures and 35 errors
```
= 5 failed, 16071 passed, 115 skipped, 1 xfailed, 2 xpassed, 9100 warnings, 35 errors in 240.50s (0:04:00) =
```
because at the two following files are missing:
```
E           FileNotFoundError: [Errno 2] No such file or directory: '/scratch/oliver/miniconda3/envs/m
daenv/lib/python3.6/site-packages/MDAnalysisTests/data/fhiaims.in'

E           FileNotFoundError: [Errno 2] No such file or directory: '/scratch/oliver/miniconda3/envs/m
daenv/lib/python3.6/site-packages/MDAnalysisTests/data/gromacs_ala10.top'

miniconda3/envs/mdaenv/lib/python3.6/bz2.py:96: FileNotFoundError
```

(One failure seems unrelated:
```
=================================== FAILURES ===================================
_______________________________ test_given_mean ________________________________
[gw2] linux -- Python 3.6.11 /scratch/oliver/miniconda3/envs/mdaenv/bin/python

pca_aligned = <MDAnalysis.analysis.pca.PCA object at 0x7fadfd89dd68>
u = <Universe with 3341 atoms>

    def test_given_mean(pca_aligned, u):
        pca = PCA(u, select=SELECTION, align=False,
                  mean=pca_aligned._mean).run()
>       assert_almost_equal(pca_aligned.cov, pca.cov, decimal=5)
E       AssertionError:
E       Arrays are not almost equal to 5 decimals
E
E       Mismatched elements: 900 / 900 (100%)
E       Max absolute difference: 1.0380358
E       Max relative difference: 106.43334522
E        x: array([[ 6.85943e-02,  6.14962e-03, -2.85235e-02, -2.46964e-03,
E                4.93089e-03, -1.45766e-02,  6.77579e-03,  6.73560e-03,
E               -6.87083e-03, -1.39547e-02,  4.34639e-03, -1.46411e-02,...
E        y: array([[ 1.10663e+00, -4.76013e-01,  2.59657e-01,  8.86885e-01,
E               -4.62178e-01,  3.13851e-01,  7.73986e-01, -4.43201e-01,
E                2.00352e-01,  4.87414e-01, -3.32454e-01,  1.01439e-01,...
```
and is not subject of this report)

## Code to reproduce the behavior ##
Install and run tests
```
pytest -n 12  --disable-pytest-warnings --pyargs MDAnalysisTests 2>&1 | tee FAILED_TESTS.log
```
(``-n 12`` is optional and needs pytest-xdist)

However, this likely affects 2.0.0-dev, too.

## Current version of MDAnalysis ##

- Which version are you using? (run `python -c "import MDAnalysis as mda; print(mda.__version__)"`)
- Which version of Python (`python -V`)?
- Which operating system?

### PR DESCRIPTION
**Pull request by lilyminium**
Fixes #2939

Changes made in this Pull Request:
 - add test file patterns in `setup.py`

I think this is why they're not included in the release. Maybe we should add a note to `datafiles.py`?

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

**Comment by orbeckst** [comment]
Thank you. Did you manually try building and running the failing tests?

File: testsuite/MDAnalysisTests/analysis/test_pca.py [hunk 1]
```diff
@@ -56,6 +56,7 @@ def pca(u):
 
 @pytest.fixture(scope='module')
 def pca_aligned(u):
+    u.transfer_to_memory()
```
Comment (lilyminium) [inline testsuite/MDAnalysisTests/analysis/test_pca.py:59]: I feel this is most efficiently done (and avoids numerous other "`transfer_to_memory()` calls) if `u` is loaded `in_memory=True` to start off with but not sure if there's a reason for this sequential nature of events

**Comment by lilyminium** [comment]
@orbeckst yes, I built the package from a new clone `python setup.py sdist` and ran tests on that. `pytest -n 12` had no errors. `python setup.py sdist` on the directory I normally work out of includes all test files in the distribution regardless of what's in `package_data` for some reason. That might be how it slipped detection in the pre-release tests.

File: testsuite/MDAnalysisTests/analysis/test_pca.py (hunk 1 above)
Comment (IAlibay) [reply testsuite/MDAnalysisTests/analysis/test_pca.py:59]: If I remember correctly, in the original issue I couldn't get the code to work even if the test was isolated. It's possible I made a mistake somewhere, but definitely would be worth testing in isolation.

**Review by orbeckst — APPROVED** [review]
From my side this looks fine. Thank you @lilyminium 

@IAlibay if you have more comments on the `transfer_to_memory()` then please block the PR.

**Review by IAlibay — CHANGES_REQUESTED** [review]
I'll temporarily block this to allow for clarification (it's possibly I'm just missing something really obvious here).

So the transfer to memory does somehow fix `test_given_mean`, but it's not exactly clear to me how/why `pca_aligned._mean` changes on alignment.

In fact, in both cases `pca._mean` is set to a NoneType? Unless I'm missing something obvious, it looks like transferring to memory just ends up providing a pre-aligned trajectory to `test_given_mean` which renders `pca._mean` obsolete?

I'm really hoping that I'm missing something obvious, but it looks to me like in all cases the `else` of this branch is never reached?
https://github.com/MDAnalysis/mdanalysis/blob/2c5e3859b9ecaa618b634b74307579f38af472c6/package/MDAnalysis/analysis/pca.py#L219-L224

This would explain why codecov has it as untested? https://codecov.io/gh/MDAnalysis/mdanalysis/src/develop/package/MDAnalysis/analysis/pca.py

**Comment by lilyminium** [comment]
As @IAlibay noted the test is a bit meaningless so I've reverted that for now and can fix in a separate PR.

**Review by IAlibay — APPROVED** [review]
lgtm

**Comment by IAlibay** [comment]
How odd, one of the travis jobs ran for over 50 mins, I've restarted it to see if it's just a bad node.

**Comment by IAlibay** [comment]
Thanks for the fix @lilyminium. Note; we'll probably also need to push this to master for 1.0.1.

**Comment by orbeckst** [comment]
I'll do the cherry-pick to master.

**Comment by orbeckst** [comment]
Fixed in 2ebb052904829c068742144d17eb38a950ed057c on master.

**Pull request merged**